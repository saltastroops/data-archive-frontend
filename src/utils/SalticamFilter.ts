export type SalticamFilter =
  | "SalticamFilter_Name"
  | "340-35"
  | "380-40"
  | "B-S1"
  | "CLR-S1"
  | "Halpha-S1"
  | "Hbn-S1"
  | "Hbw-S1"
  | "I-S1"
  | "R-S1"
  | "Sb-S1"
  | "SDSSg-S1"
  | "SDSSi-S1"
  | "SDSSr-S1"
  | "SDSSu-S1"
  | "SDSSz-S1"
  | "SR613-21"
  | "SR708-25"
  | "SR815-29"
  | "SR862-32"
  | "Su-S1"
  | "Sv-S1"
  | "Sy-S1"
  | "U-S1"
  | "V-S1";

export interface ISalticamFilter {
  name: SalticamFilter;
  descriptiveName: string;
}
