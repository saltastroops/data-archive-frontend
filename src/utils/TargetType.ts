export type TargetType = string;

export const GALAXY = "10";

export const ISM = "20";

export const SOLAR_SYSTEM_BODY = "30";

export const STAR = "40";
