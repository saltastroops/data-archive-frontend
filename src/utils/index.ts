export const toTitleCase = (s: string) => {
  s = s.replace(/_/g, " ");
  return s.replace(/\w\S*/g, (txt: string) => {
    return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();
  });
};
