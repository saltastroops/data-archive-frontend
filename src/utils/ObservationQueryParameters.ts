// TypeScript interfaces describing the state of the search form

import { SalticamFilter } from "./SalticamFilter";
import { TargetType } from "./TargetType";

/**
 * An interface describing the React state of the observation query form.
 *
 * This state is used for populating all search form fields (and error messages)
 * and is parsed into the GraphQL query for searching observations.
 *
 * Properties:
 * -----------
 * general:
 *     General information about the observation.
 * target:
 *     Target details.
 * telescope:
 *     Telescope (and instrument) details.
 */
// TODO: Remove loading, replace with IObservationQueryParameters
export interface ISearchFormState {
  general: IGeneral;
  target: ITarget;
  telescope?: ITelescope;
  loading: boolean;
}

/**
 * The available calibration types.
 */
export type CalibrationType = "arc" | "bias" | "flat" | "standard";

/**
 * An interface for query parameters related to general information.
 *
 * Properties:
 * -----------
 * errors:
 *     Error messages.
 * observationNight:
 *     Date of the night when the observation was taken. This
 *     refers to the date whe the night started.
 * proposalCode:
 *     Identifier of the proposal to which the observation belongs.
 * proposalTitle:
 *     Title of the proposal to which the observation belongs.
 * calibrations:
 *     Calibration types to include in the search results
 *
 */
export interface IGeneralErrors {
  observationNight?: string;
  principalInvestigator?: string;
  proposalCode?: string;
  proposalTitle?: string;
  calibrations?: string;
}
export interface IGeneral {
  errors: IGeneralErrors;
  observationNight?: string;
  principalInvestigator?: string;
  proposalCode?: string;
  proposalTitle?: string;
  calibrations: Set<CalibrationType>;
}

/**
 * The available target name resolvers.
 */
export type TargetResolver = "Simbad" | "NED" | "VizieR";

/**
 * The available units for the search cone radius.
 */
export type SearchConeRadiusUnits = "arcseconds" | "arcminutes" | "degrees";

/**
 * An interface for query parameters related to target details.
 *
 * Properties:
 * -----------
 * declination:
 *     Declination.
 * errors:
 *     Errors.
 * name:
 *     Target name.
 * resolver:
 *     Resolver to use for resolving the target name to a position.
 * rightAscension:
 *     Right ascension.
 * searchConeRadius:
 *     Radius of the search cone radius, in the units given by
 *     searchConeRadiusUnits.
 * searchConeRadiusUnits:
 *     Units of the search cone radius given by searchConeRadius.
 *
 */
export interface ITarget {
  declination?: string;
  errors: ITargetErrors;
  loading?: string;
  name?: string;
  resolver: TargetResolver;
  rightAscension?: string;
  searchConeRadius?: string;
  searchConeRadiusUnits: SearchConeRadiusUnits;
  targetTypes: Set<TargetType>;
}

export interface ITargetErrors {
  name?: string;
  declination?: string;
  resolver?: string;
  rightAscension?: string;
  searchConeRadius?: string;
  searchConeRadiusUnits?: string;
  targetTypes?: string;
}

// TELESCOPES

/**
 * The supported telescope names.
 */
export type TelescopeName = "Lesedi" | "SALT" | "1.9 m";

/**
 * An interface for query parameters related to a telescope.
 *
 * This serves as a placeholder for more concrete interfaces such as ISALT.
 *
 * Properties:
 * -----------
 * name:
 *     Telescope name.
 */
export interface ITelescope {
  name?: TelescopeName;
}

/**
 * An interface for query parameters related to SALT.
 *
 * Properties:
 * -----------
 * instrument:
 *     A SALT instrument.
 * name:
 *     The string "SALT".
 */
export interface ISALT extends ITelescope {
  instrument: IInstrument;
  name: "SALT";
}

/**
 * An interface for query parameters related to Lesedi.
 *
 * Properties:
 * -----------
 * instrument:
 *     An instrument used on Lesedi.
 * name:
 *     The string "Lesedi".
 */
export interface ILesedi extends ITelescope {
  instrument: IInstrument;
  name: "Lesedi";
}

/**
 * An interface for query parameters related to the 1.9 m Telescope.
 *
 * Properties:
 * -----------
 * instrument:
 *     An instrument used on the 1./9 m Telescope.
 * name:
 *     The string "1.9 m".
 */
export interface IOneDotNineM extends ITelescope {
  instrument: IInstrument;
  name: "1.9 m";
}

// INSTRUMENTS

/**
 * The available instrument names.
 */
export type InstrumentName =
  | "Salticam"
  | "RSS"
  | "HRS"
  | "BVIT"
  | "SpUpNIC"
  | "SHOC"
  | "HIPPO";

/**
 * An interface for query parameters related to an instrument.
 *
 * This serves as a placeholder for more concrete interfaces such as ISalticam.
 *
 * Properties:
 * -----------
 * name:
 *     Telescope name.
 */
export interface IInstrument {
  name: InstrumentName;
}

/**
 * An interface for query parameters related to Salticam.
 *
 * Properties:
 * -----------
 * detectorMode:
 *     Detector mode.
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * filter:
 *     Filter.
 * name:
 *     The string "Salticam".
 */
export interface ISalticam extends IInstrument {
  detectorMode?: "Normal" | "Slot Mode";
  errors: {
    detectorMode?: string;
    exposureTime?: string;
    filter?: string;
  };
  exposureTime?: string;
  filter?: SalticamFilter;
  name: "Salticam";
}

/**
 * An interface for query parameters related to RSS.
 *
 * Properties:
 * -----------
 * detectorMode:
 *     Detector mode.
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * filter:
 *     Filter.
 * mode:
 *     Instrument mode specific details.
 * name:
 *     The string "RSS".
 */
export interface IRSS extends IInstrument {
  detectorMode?: RSSDetectorMode;
  errors: {
    detectorMode?: string;
    exposureTime?: string;
    filter?: string;
  };
  exposureTime?: string;
  filter?: string;
  modes?: IRSSModes;
  name: "RSS";
}

export interface IRSSModes {
  errors: {
    fabryPerotMode?: string;
    grating?: string;
    polarimetryModes?: string;
  };
  fabryPerotMode?: RSSFabryPerotMode;
  grating?: RSSGrating;
  names: Set<RSSInstrumentMode>;
  polarimetryModes?: Set<RSSPolarimetryMode>;
}

export type RSSDetectorMode = "Normal" | "Slot Mode" | "";

export type RSSGrating =
  | "Open"
  | "pg0300"
  | "pg0900"
  | "pg1300"
  | "pg1800"
  | "pg2300"
  | "pg3000"
  | "";

export type RSSInstrumentMode =
  | "Fabry Perot"
  | "FP polarimetry"
  | "Imaging"
  | "Polarimetric imaging"
  | "MOS"
  | "MOS polarimetry"
  | "Spectropolarimetry"
  | "Spectroscopy";

export type RSSFabryPerotMode = "HR" | "LR" | "MR" | "TF" | "";

export type RSSPolarimetryMode =
  | "All Stokes"
  | "Circular"
  | "Linear"
  | "Linear Hi"
  | "";

/**
 * An interface for query parameters related to HRS.
 *
 * Properties:
 * -----------
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * mode:
 *     Resolution mode.
 * name:
 *     The string "HRS".
 */
export interface IHRS extends IInstrument {
  errors: {
    exposureTime?: string;
    mode?: string;
  };
  exposureTime?: string;
  mode?: HRSMode;
  name: "HRS";
}

export type HRSMode =
  | "High Resolution"
  | "High Stability"
  | "Int Cal Fibre"
  | "Low Resolution"
  | "Medium Resolution";

export interface IBVIT extends IInstrument {
  errors: {};
  filter?: BVITFilter;
  mode?: "Imaging" | "Streaming";
  name: "BVIT";
}

export type BVITFilter = "B" | "H-alpha" | "Open" | "R" | "U" | "V";

/**
 * An interface for query parameters related to HIPPO.
 *
 * Properties:
 * -----------
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * filter:
 *     Filter.
 * name:
 *     The string "HIPPO".
 */
export interface IHIPPO extends IInstrument {
  errors: {
    exposureTime?: string;
    filter?: string;
  };
  exposureTime?: string;
  filter?: string;
  name: "HIPPO";
}

/**
 * An interface for query parameters related to SHOC.
 *
 * Properties:
 * -----------
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * filter:
 *     Filter.
 * name:
 *     The string "SHOC".
 */
export interface ISHOC extends IInstrument {
  errors: {
    exposureTime?: string;
    filter?: string;
  };
  exposureTime?: string;
  filter?: string;
  name: "SHOC";
}

/**
 * An interface for query parameters related to SpUpNIC.
 *
 * Properties:
 * -----------
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * filter:
 *     Filter.
 * name:
 *     The string "SpUpNIC".
 */
export interface ISpUpNIC extends IInstrument {
  errors: {
    exposureTime?: string;
    filter?: string;
  };
  exposureTime?: string;
  filter?: string;
  name: "SpUpNIC";
}

export interface IFile {
  id: string;
  filename: string;
  name: string;
  dataType: string;
  isReduced: boolean;
  targetName: string;
  rightAscension: number;
  declination: number;
  observationNight: string;
  category: string;
  telescope: string;
  instrument: string;
  observationId?: string;
  url?: string;
}
export interface IObservation {
  id: string;
  name: string | "Unknown Observation";
  proposal?: string;
  telescope: string;
  startTime: string;
  files: IFile[];
}
