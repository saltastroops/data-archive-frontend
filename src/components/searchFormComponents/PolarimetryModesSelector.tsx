import * as React from "react";
import styled from "styled-components";
import { RSSPolarimetryMode } from "../../utils/ObservationQueryParameters";
import { LargeCheckbox } from "../basicComponents/LargeCheckbox";

interface IPolarimetryModesSelectorProps {
  onChange: (polarimetryModes: Set<RSSPolarimetryMode>) => void;
  polarimetryModes?: Set<RSSPolarimetryMode>;
}

const PolarimetryModesDiv = styled.div`
  && {
    margin-top: 5px;
  }

  label {
    padding-right: 10px;
  }
`;

/**
 * A component for polarimetry modes.
 *
 * The component does not keep internal state, so it is up to the parent
 * component to ensure that the polarimetry modes passed as a property remain
 * up-to-date.
 *
 * Properties:
 * -----------
 * polarimetryModes:
 *     The selected polarimetry modes.
 * onChange:
 *     The function to call when the user selects or unselects a polarimetry
 *     mode, which should take the selected target types as its single
 *     argument.
 */
export default class PolarimetryModesSelector extends React.Component<
  IPolarimetryModesSelectorProps
> {
  onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const polarimetryModes =
      this.props.polarimetryModes || new Set<RSSPolarimetryMode>();
    const updatedPolarimetryModes = new Set<RSSPolarimetryMode>(
      polarimetryModes
    );
    if (e.target.checked) {
      updatedPolarimetryModes.add(e.target.value as RSSPolarimetryMode);
    } else {
      updatedPolarimetryModes.delete(e.target.value as RSSPolarimetryMode);
    }

    this.props.onChange(updatedPolarimetryModes);
  };

  render() {
    const polarimetryModes =
      this.props.polarimetryModes || new Set<RSSPolarimetryMode>();

    return (
      <PolarimetryModesDiv>
        <label>
          <LargeCheckbox
            checked={polarimetryModes.has("Linear")}
            data-test="linear"
            onChange={this.onChange}
            value={"Linear"}
          />
          Linear
        </label>
        <label>
          <LargeCheckbox
            checked={polarimetryModes.has("Linear Hi")}
            data-test="linear-hi"
            onChange={this.onChange}
            value={"Linear Hi"}
          />
          Linear Hi
        </label>
        <label>
          <LargeCheckbox
            checked={polarimetryModes.has("Circular")}
            data-test="circular"
            onChange={this.onChange}
            value={"Circular"}
          />
          Circular
        </label>
        <label>
          <LargeCheckbox
            checked={polarimetryModes.has("All Stokes")}
            data-test="all-stokes"
            onChange={this.onChange}
            value={"All Stokes"}
          />
          All Stokes
        </label>
      </PolarimetryModesDiv>
    );
  }
}
