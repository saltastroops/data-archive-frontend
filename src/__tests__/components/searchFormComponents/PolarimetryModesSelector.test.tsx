import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import wait from "waait";
import PolarimetryModesSelector from "../../../components/searchFormComponents/PolarimetryModesSelector";
import { RSSPolarimetryMode } from "../../../utils/ObservationQueryParameters";

describe("PolarimetryModesSelector", () => {
  it("should render correctly with no polarimetry modes", () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <PolarimetryModesSelector
        onChange={onChange}
        polarimetryModes={new Set<RSSPolarimetryMode>()}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render correctly with polarimetry modes", () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <PolarimetryModesSelector
        onChange={onChange}
        polarimetryModes={new Set<RSSPolarimetryMode>(["Linear", "Circular"])}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should call the onChange method with the correct argument when adding a polarimetry mode", async () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <PolarimetryModesSelector
        onChange={onChange}
        polarimetryModes={new Set<RSSPolarimetryMode>(["Linear", "Circular"])}
      />
    );
    const linearHiCheckbox = wrapper.find('input[data-test="linear-hi"]');
    linearHiCheckbox.simulate("change", {
      target: { checked: true, value: "Linear Hi" }
    });
    await wait(0);
    expect(onChange).toHaveBeenCalledWith(
      new Set<RSSPolarimetryMode>(["Linear", "Circular", "Linear Hi"])
    );
  });

  it("should call the onChange method with the correct argument when removing a polarimetry mode", async () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <PolarimetryModesSelector
        onChange={onChange}
        polarimetryModes={new Set<RSSPolarimetryMode>(["Linear", "Circular"])}
      />
    );
    const circularCheckbox = wrapper.find('input[data-test="circular"]');
    circularCheckbox.simulate("change", {
      target: { checked: false, value: "Circular" }
    });
    await wait(0);
    expect(onChange).toHaveBeenCalledWith(
      new Set<RSSPolarimetryMode>(["Linear"])
    );
  });
});
