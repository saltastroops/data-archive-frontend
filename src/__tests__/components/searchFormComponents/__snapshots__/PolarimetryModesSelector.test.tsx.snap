// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PolarimetryModesSelector should render correctly with no polarimetry modes 1`] = `
.c1.c1 {
  width: 18px;
  height: 18px;
}

.c0.c0 {
  margin-top: 5px;
}

.c0 label {
  padding-right: 10px;
}

<PolarimetryModesSelector
  onChange={[MockFunction]}
  polarimetryModes={Set {}}
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  && {
    margin-top: 5px;
  }

  label {
    padding-right: 10px;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <label>
          <styled.input
            checked={false}
            data-test="linear"
            onChange={[Function]}
            value="Linear"
          >
            <StyledComponent
              checked={false}
              data-test="linear"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [
                    Object {
                      "className": "checkbox",
                      "type": "checkbox",
                    },
                  ],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": true,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  && {
    width: 18px;
    height: 18px;
  }
",
                    ],
                  },
                  "displayName": "styled.input",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onChange={[Function]}
              value="Linear"
            >
              <input
                checked={false}
                className="checkbox c1"
                data-test="linear"
                onChange={[Function]}
                type="checkbox"
                value="Linear"
              />
            </StyledComponent>
          </styled.input>
          Linear
        </label>
        <label>
          <styled.input
            checked={false}
            data-test="linear-hi"
            onChange={[Function]}
            value="Linear Hi"
          >
            <StyledComponent
              checked={false}
              data-test="linear-hi"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [
                    Object {
                      "className": "checkbox",
                      "type": "checkbox",
                    },
                  ],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": true,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  && {
    width: 18px;
    height: 18px;
  }
",
                    ],
                  },
                  "displayName": "styled.input",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onChange={[Function]}
              value="Linear Hi"
            >
              <input
                checked={false}
                className="checkbox c1"
                data-test="linear-hi"
                onChange={[Function]}
                type="checkbox"
                value="Linear Hi"
              />
            </StyledComponent>
          </styled.input>
          Linear Hi
        </label>
        <label>
          <styled.input
            checked={false}
            data-test="circular"
            onChange={[Function]}
            value="Circular"
          >
            <StyledComponent
              checked={false}
              data-test="circular"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [
                    Object {
                      "className": "checkbox",
                      "type": "checkbox",
                    },
                  ],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": true,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  && {
    width: 18px;
    height: 18px;
  }
",
                    ],
                  },
                  "displayName": "styled.input",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onChange={[Function]}
              value="Circular"
            >
              <input
                checked={false}
                className="checkbox c1"
                data-test="circular"
                onChange={[Function]}
                type="checkbox"
                value="Circular"
              />
            </StyledComponent>
          </styled.input>
          Circular
        </label>
        <label>
          <styled.input
            checked={false}
            data-test="all-stokes"
            onChange={[Function]}
            value="All Stokes"
          >
            <StyledComponent
              checked={false}
              data-test="all-stokes"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [
                    Object {
                      "className": "checkbox",
                      "type": "checkbox",
                    },
                  ],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": true,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  && {
    width: 18px;
    height: 18px;
  }
",
                    ],
                  },
                  "displayName": "styled.input",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onChange={[Function]}
              value="All Stokes"
            >
              <input
                checked={false}
                className="checkbox c1"
                data-test="all-stokes"
                onChange={[Function]}
                type="checkbox"
                value="All Stokes"
              />
            </StyledComponent>
          </styled.input>
          All Stokes
        </label>
      </div>
    </StyledComponent>
  </styled.div>
</PolarimetryModesSelector>
`;

exports[`PolarimetryModesSelector should render correctly with polarimetry modes 1`] = `
.c1.c1 {
  width: 18px;
  height: 18px;
}

.c0.c0 {
  margin-top: 5px;
}

.c0 label {
  padding-right: 10px;
}

<PolarimetryModesSelector
  onChange={[MockFunction]}
  polarimetryModes={
    Set {
      "Linear",
      "Circular",
    }
  }
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  && {
    margin-top: 5px;
  }

  label {
    padding-right: 10px;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <label>
          <styled.input
            checked={true}
            data-test="linear"
            onChange={[Function]}
            value="Linear"
          >
            <StyledComponent
              checked={true}
              data-test="linear"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [
                    Object {
                      "className": "checkbox",
                      "type": "checkbox",
                    },
                  ],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": true,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  && {
    width: 18px;
    height: 18px;
  }
",
                    ],
                  },
                  "displayName": "styled.input",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onChange={[Function]}
              value="Linear"
            >
              <input
                checked={true}
                className="checkbox c1"
                data-test="linear"
                onChange={[Function]}
                type="checkbox"
                value="Linear"
              />
            </StyledComponent>
          </styled.input>
          Linear
        </label>
        <label>
          <styled.input
            checked={false}
            data-test="linear-hi"
            onChange={[Function]}
            value="Linear Hi"
          >
            <StyledComponent
              checked={false}
              data-test="linear-hi"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [
                    Object {
                      "className": "checkbox",
                      "type": "checkbox",
                    },
                  ],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": true,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  && {
    width: 18px;
    height: 18px;
  }
",
                    ],
                  },
                  "displayName": "styled.input",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onChange={[Function]}
              value="Linear Hi"
            >
              <input
                checked={false}
                className="checkbox c1"
                data-test="linear-hi"
                onChange={[Function]}
                type="checkbox"
                value="Linear Hi"
              />
            </StyledComponent>
          </styled.input>
          Linear Hi
        </label>
        <label>
          <styled.input
            checked={true}
            data-test="circular"
            onChange={[Function]}
            value="Circular"
          >
            <StyledComponent
              checked={true}
              data-test="circular"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [
                    Object {
                      "className": "checkbox",
                      "type": "checkbox",
                    },
                  ],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": true,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  && {
    width: 18px;
    height: 18px;
  }
",
                    ],
                  },
                  "displayName": "styled.input",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onChange={[Function]}
              value="Circular"
            >
              <input
                checked={true}
                className="checkbox c1"
                data-test="circular"
                onChange={[Function]}
                type="checkbox"
                value="Circular"
              />
            </StyledComponent>
          </styled.input>
          Circular
        </label>
        <label>
          <styled.input
            checked={false}
            data-test="all-stokes"
            onChange={[Function]}
            value="All Stokes"
          >
            <StyledComponent
              checked={false}
              data-test="all-stokes"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [
                    Object {
                      "className": "checkbox",
                      "type": "checkbox",
                    },
                  ],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": true,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  && {
    width: 18px;
    height: 18px;
  }
",
                    ],
                  },
                  "displayName": "styled.input",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onChange={[Function]}
              value="All Stokes"
            >
              <input
                checked={false}
                className="checkbox c1"
                data-test="all-stokes"
                onChange={[Function]}
                type="checkbox"
                value="All Stokes"
              />
            </StyledComponent>
          </styled.input>
          All Stokes
        </label>
      </div>
    </StyledComponent>
  </styled.div>
</PolarimetryModesSelector>
`;
